src/main/sensors/gyro.c

```diff
@@ -454,8 +478,80 @@ bool gyroInit(void)
         break;
     }
     firstArmingCalibrationWasStarted = false;
+
+    bool ret = false;
     memset(&gyro, 0, sizeof(gyro));
-    return gyroInitSensor(&gyroSensor1);
+    gyroToUse = gyroConfig()->gyro_to_use;
+
+#if defined(USE_DUAL_GYRO) && defined(GYRO_1_CS_PIN)
+    if (gyroToUse == GYRO_CONFIG_USE_GYRO_1 || gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH) {
+        gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin = IOGetByTag(IO_TAG(GYRO_1_CS_PIN));
+        IOInit(gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin, OWNER_MPU_CS, 0);
+        IOHi(gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin); // Ensure device is disabled, important when two devices are on the same bus.
+        IOConfigGPIO(gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin, SPI_IO_CS_CFG);
+    }
+#endif
+
+#if defined(USE_DUAL_GYRO) && defined(GYRO_2_CS_PIN)
+    if (gyroToUse == GYRO_CONFIG_USE_GYRO_2 || gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH) {
+        gyroSensor2.gyroDev.bus.busdev_u.spi.csnPin = IOGetByTag(IO_TAG(GYRO_2_CS_PIN));
+        IOInit(gyroSensor2.gyroDev.bus.busdev_u.spi.csnPin, OWNER_MPU_CS, 0);
+        IOHi(gyroSensor2.gyroDev.bus.busdev_u.spi.csnPin); // Ensure device is disabled, important when two devices are on the same bus.
+        IOConfigGPIO(gyroSensor2.gyroDev.bus.busdev_u.spi.csnPin, SPI_IO_CS_CFG);
+    }
+#endif
+
+    gyroSensor1.gyroDev.gyroAlign = ALIGN_DEFAULT;
+
+#if defined(GYRO_1_EXTI_PIN)
+    gyroSensor1.gyroDev.mpuIntExtiTag =  IO_TAG(GYRO_1_EXTI_PIN);
+#elif defined(MPU_INT_EXTI)
+    gyroSensor1.gyroDev.mpuIntExtiTag =  IO_TAG(MPU_INT_EXTI);
+#elif defined(USE_HARDWARE_REVISION_DETECTION)
+    gyroSensor1.gyroDev.mpuIntExtiTag =  selectMPUIntExtiConfigByHardwareRevision();
+#else
+    gyroSensor1.gyroDev.mpuIntExtiTag =  IO_TAG_NONE;
+#endif // GYRO_1_EXTI_PIN
+#ifdef USE_DUAL_GYRO
+#ifdef GYRO_1_ALIGN
+    gyroSensor1.gyroDev.gyroAlign = GYRO_1_ALIGN;
+#endif
+    gyroSensor1.gyroDev.bus.bustype = BUSTYPE_SPI;
+    spiBusSetInstance(&gyroSensor1.gyroDev.bus, GYRO_1_SPI_INSTANCE);
```

Comment:
This is causing a compilation error in KAKUTEF4.

Reply:
fixed.